test/RoboCop.ts

```diff
@@ -132,32 +133,37 @@ describe("RoboCop", () => {
       await network.provider.send("evm_setAutomine", [false]);
       const tx1 = await roboCop.connect(ruleMakerWallet).createRule([passingTrigger], [executableAction]);
       const tx2 = await roboCop.connect(ruleMakerWallet).createRule([passingTrigger], [executableAction]);
-      // different user, so this 3rd rule should work
-      const tx3 = await roboCop.connect(ruleMakerWallet).createRule([passingTrigger], [executableAction]);
+
       await network.provider.send("evm_mine", []);
       await network.provider.send("evm_setAutomine", [true]);
+      // different block time, so this 3rd rule should work
+      await expect(roboCop.connect(ruleMakerWallet).createRule([passingTrigger], [executableAction])).to.emit(
+        roboCop,
+        "Created"
+      );
 
       try {
         await tx1.wait();
       } catch {
         expect.fail();
       }
 
-      try {
-        await tx2.wait();
-        // cant figure how to confirm whether the error is a duplicate rule error.
-        // this will suffice for now.
-        expect.fail();
-      } catch (err) {
-        /* pass */
-      }
-
-      try {
-        await tx3.wait();
-      } catch (err) {
-        // this acts as a control to ensure the error wasnt due to the evm_mine stuff
-        expect.fail();
-      }
+      const trace = await network.provider.send("debug_traceTransaction", [
+        tx2.hash,
+        {
+          disableMemory: true,
+          disableStack: true,
+          disableStorage: true,
```

Comment:
why do we need these flags? the tests seem to pass without them as well

Reply:
Those flags r unnecessary, shall remove them